internal/api/v1/services/instance.go

```diff
@@ -169,16 +169,37 @@ func (s *Instance) GetInstancesByJobID(ctx context.Context, jobID uint) ([]model
 
 // Terminate handles the termination of instances for a given job name and instance names.
 func (s *Instance) Terminate(ctx context.Context, ownerID uint, jobName string, instanceNames []string) error {
+	// First verify the job exists and belongs to the owner
 	job, err := s.jobService.jobRepo.GetByName(ctx, ownerID, jobName)
 	if err != nil {
 		return fmt.Errorf("failed to get job: %w", err)
 	}
 
-	instances, err := s.repo.GetByNames(ctx, instanceNames)
+	// Get instances that belong to this job and match the provided names
+	instances, err := s.repo.GetByJobIDAndNames(ctx, job.ID, instanceNames)
 	if err != nil {
 		return fmt.Errorf("failed to get instances: %w", err)
 	}
 
+	// Verify we found all requested instances
+	if len(instances) == 0 {
+		return fmt.Errorf("no active instances found with the specified names for job '%s'", jobName)
+	}
+	if len(instances) != len(instanceNames) {
```

Comment:
i dont think we need this.  a user should be able to submit the same delete request multiple times and it is just a no-op if things were already deleted.  

logging that info is fine, but i dont think it should be an error